lib/strategies/webapp-strategy.js

```diff
@@ -273,6 +275,12 @@ function redirectForAuthorization(req, options, strategy) {
 		return;
 	}
 
+	var stateParameter = { anonymousLogin: false, state: TokenUtil.getRandomNumber()};
+	if (options.allowAnonymousLogin === true) {
```

Comment:
I don't think this works like this...Can't you log in through facebook/etc. with this set to true?